### PR TITLE
Add support for working hours. Fixes #3

### LINKED ISSUE #3: Make it possible to configure Working Hours and hide the review count outside them
It'd be lovely if MyQOnly didn't try to guilt me into doing reviews during my evenings and weekends.

### PR DESCRIPTION
**Pull request by 6a68**
* Add day and time pickers to the options page

* Clear the review badge if the update interval runs outside the working hours

Here's the options page with the UI as implemented:

screen shot 2018-10-18 at 12 33 16 pm

### CODE REVIEW

File: content/options.html [hunk 1]
```diff
@@ -22,7 +22,45 @@
 
   <label for="github-username">Github Username</label>
   <input type="text" id="github-username" data-type="ghuser">
+
+  <h1>Working Hours</h1>
+  <br>
+
+  <label for="working-hours-checkbox">Only show review count during working hours</label>
+  <input type="checkbox" id="working-hours-checkbox">
```
Comment (mikeconley) [inline content/options.html]: Could you please put the checkbox before the label? Perhaps even inside the <label> itself.

File: content/options.html [hunk 2]
```diff
@@ -22,7 +22,45 @@
...
+
+  <fieldset id="working-hours" disabled>
+    <label for="start-time">Start time</label>
+    <input type="time" id="start-time" value="09:00">
+    <br>
+    <label for="end-time">End time</label>
+    <input type="time" id="end-time" value="17:00">
```
Comment (mikeconley) [inline content/options.html]: It doesn't need to be fancy, but it'd be lovely if these could be horizontally aligned with one another. Perhaps using CSS grid, like I did for the other fields above?

File: content/options.html [hunk 3]
```diff
@@ -22,7 +22,45 @@
...
+
+    <div id="days">
+      <p>Working Days:</p>
+      <label for="sunday">Sunday</label>
```
Comment (mikeconley) [inline content/options.html]: Let's put the checkboxes before the labels here too. If they can be lined up properly, so much the better.

File: content/options.html [hunk 4]
```diff
@@ -22,7 +22,45 @@
...
   <label for="github-username">Github Username</label>
   <input type="text" id="github-username" data-type="ghuser">
+
+  <h1>Working Hours</h1>
+  <br>
+
+  <label for="working-hours-checkbox">Only show review count during working hours</label>
+  <input type="checkbox" id="working-hours-checkbox">
+
+  <fieldset id="working-hours" disabled>
```
Comment (mikeconley) [inline content/options.html]: I completely forgot that fieldsets could be mass enabled / disabled like this! Nice!

File: content/options.js [hunk 5]
```diff
@@ -20,6 +20,34 @@ const Options = {
...
+
+    document.querySelector("#working-hours-checkbox").checked = workingHours.enabled;
+    if (workingHours.enabled) {
+      document.querySelector("#working-hours").removeAttribute("disabled");
+    } else {
+      document.querySelector("#working-hours").setAttribute("disabled", "disabled");
+    }
+    document.querySelector("#start-time").value  = workingHours.startTime;
+    document.querySelector("#end-time").value    = workingHours.endTime;
+    document.querySelector("#sunday").checked    = workingHours.days.includes("sunday");
```
Comment (mikeconley) [inline content/options.js]: I think I might prefer if we query the document for the days, and then loop over the nodes, setting the checked value if the IDs (or maybe better, if a data-day attribute) is included in workingHours.days.

File: background.js [hunk 6]
```diff
@@ -137,11 +137,84 @@ const MyQOnly = {
...
+      let endTime = new Date();
+      let [endHours, endMinutes] = workingHours.endTime.split(":");
+      endTime.setHours(endHours, endMinutes);
+      if (currentTime > endTime) {
+        console.log(`Current time (${currentTime.toLocaleTimeString()}) is later than end time (${endTime.toLocaleTimeString()})`);
+        return false;
+      }
+    }
+
+    // Unlike the times, workingHours.days should never be falsy: the days are
```
Comment (mikeconley) [inline background.js]: Typo: "falsy" -> "false-y"

Comment (6a68) [reply background.js]: Hmm, this is an interesting point. I've only ever seen `falsy`, and there is https://mdn.io/glossary/falsy, but OTOH, Google surprisingly turns up a ton of refs for `falsey`, and https://english.stackexchange.com/questions/109996/is-it-falsy-or-falsey definitely fails to resolve the ambiguity. But, hey, your project, your spelling :+1:

**Comment by 6a68** [comment]
Thanks for the quick review. Should get these updates in tomorrow at lunch, I reckon

**Comment by Callek** [comment]
Drive-By:

Day of week information is very locale specific and can vary, e.g. first day isn't always Sunday. -- We have docs for this https://firefox-source-docs.mozilla.org/intl/dataintl.html?highlight=mozintl#mozintl-getcalendarinfo-locale

I'm not sure if it's available to web extensions though (nor am I sure if we care about it for the purposes of this extension) so I'm just raising it here. :Gandalf would know if it is available to web extensions.

**Comment by 6a68** [comment]
@mikeconley Looks like this might not happen as soon as I'd hoped. Are you still up for doing the final tweaks?

**Comment by mikeconley** [comment]
Sure thing - I can take this over the line. Thanks for the start!

**Comment by mikeconley** [comment]
Merged without addressing the original feedback. I'll apply the feedback later.

**Pull request merged**